src/streamlink/plugins/dlive.py

```diff
@@ -0,0 +1,390 @@
+
+                fragment StreamChatroomInputFrag on User
+                {
+                    chatMode
+                    chatInterval
+                    myRoomRole @include(if: $isLoggedIn)
+                    livestream
+                    {
+                        permlink
+                        creator {      username      __typename    }
+                        __typename
+                    }
+                    ...StreamChatMemberManageTabFrag
+                    ...StreamChatModeSettingsFrag
+                    ...EmoteBoardStreamerFrag
+                    __typename
+                }
+
+                fragment StreamChatMemberManageTabFrag on User
+                {
+                    id
+                    username
+                    displayname
+                    myRoomRole @include(if: $isLoggedIn)
+                    __typename
+                }
+
+                fragment StreamChatModeSettingsFrag on User
+                {
+                    id
+                    chatMode
+                    allowEmote
+                    chatInterval
+                    __typename
+                }
+
+                fragment EmoteBoardStreamerFrag on User
+                {
+                    id
+                    username
+                    partnerStatus
+                    myRoomRole @include(if: $isLoggedIn)
+                    emote @include(if: $isLoggedIn)
+                    {
+                        channel
+                        {
+                            list
+                            {
+                                name
+                                username
+                                sourceURL
+                                mimeType
+                                level
+                                type
+                                __typename
+                            }
+                            __typename
+                        }
+                        __typename
+                    }
+                    __typename
+                }
+
+                fragment VStreamChatSenderInfoFrag on SenderInfo
+                {
+                    subscribing
+                    role
+                    roomRole
+                    sender
+                    {
+                        id
+                        username
+                        displayname
+                        avatar
+                        partnerStatus
+                        __typename
+                    }
+                __typename
+                }
+                """
+        }
+        res = self.session.http.post(reqUrl, json=jsonData)
+
+        jsonres = self.session.http.json(res, schema=self._INFO_SCHEMA)
+
+        hls_url = "https://live.prd.dlive.tv/hls/live/" \
+            + jsonres["data"]["userByDisplayName"]["username"] + ".m3u8"
+
+        try:
+            return HLSStream.parse_variant_playlist(self.session, hls_url)
+        except Exception:
```

Comment:
don't use an empty `Exception` or an Exception at all here

Reply:
The problem I had was that the user will see an exception if the server returns an error code if the playlist does not exist because the livestream is offline. I copied this one from another plugin... What would be the better solution here?

Reply:
> What would be the better solution here?

getting the online/offline status from `__APOLLO_STATE__`